instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/http/HttpClientAttributesExtractor.java

```diff
@@ -65,6 +72,9 @@ public static <REQUEST, RESPONSE> HttpClientAttributesExtractorBuilder<REQUEST,
     internalNetworkExtractor = builder.buildNetworkExtractor();
     internalServerExtractor = builder.buildServerExtractor();
     resendCountIncrementer = builder.resendCountIncrementer;
+    redactSensitiveParameters =
+        Boolean.getBoolean(
+            "otel.instrumentation.http.client.experimental.redact-sensitive-parameters");
```

Comment:
This does not align with our usual way of handling configuration options. It only supports system properties while we usually also accept environment variables. Perhaps it would be better to handle it the same way as other http client experimental options in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/ed0fb2e97aa4ec3742d0264864fdfccb59ccffae/instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/builder/internal/DefaultHttpClientInstrumenterBuilder.java#L243 so that it could also be used from spring boot configuration. Since the `HttpClientAttributesExtractor` and builder are stable introducing experimental options will require trickery with internal classes.